### PR TITLE
OkHttp for courier strategies

### PR DESCRIPTION
**Pull request by norkator**
All of them are tested except GLS

### CODE REVIEW

File: app/src/main/java/com/nitramite/courier/ArraPakettiStrategy.java [hunk 1]
```diff
@@ -2,88 +2,56 @@
...
-                    public void checkServerTrusted(
-                            java.security.cert.X509Certificate[] certs, String authType) {
-                    }
-                }
-        };
         try {
-            SSLContext sc = SSLContext.getInstance("SSL");
-            sc.init(null, trustAllCerts, new java.security.SecureRandom());
-            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
             String url = "https://www.r-kioski.fi/wordpress/wp-admin/admin-ajax.php";
```
Comment (ristomatti) [inline app/src/main/java/com/nitramite/courier/ArraPakettiStrategy.java:34]: Unrelated but maybe add this as a constant for tidyness?

File: app/src/main/java/com/nitramite/courier/CainiaoStrategy.java [hunk 2]
```diff
@@ -21,75 +21,30 @@
...
-        while ((inputLine = in.readLine()) != null) {
-            response.append(inputLine);
-        }
-        in.close();
-        return response.toString();
+    private String getRequest(String url) throws IOException {
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url(url)
+                .addHeader("User-Agent", "Mozilla/5.0")
```
Comment (ristomatti) [inline app/src/main/java/com/nitramite/courier/CainiaoStrategy.java:44]: Is there a particular reason for this UA string to be different than the one for `ArraPakettiStrategy`? Could it be a constant somew

File: app/src/main/java/com/nitramite/courier/CainiaoStrategy.java [hunk 3]
```diff
@@ -21,75 +21,30 @@
...
-        con.setRequestProperty("User-Agent", USER_AGENT);
-        BufferedReader in = new BufferedReader(
-                new InputStreamReader(con.getInputStream()));
-        String inputLine;
-        StringBuilder response = new StringBuilder();
-        while ((inputLine = in.readLine()) != null) {
-            response.append(inputLine);
-        }
-        in.close();
-        return response.toString();
```
Comment (ristomatti) [inline app/src/main/java/com/nitramite/courier/CainiaoStrategy.java:92]: This new code looks a lot cleaner! :+1:

File: app/src/main/java/com/nitramite/courier/DHLActiveTrackingStrategy.java [hunk 4]
```diff
@@ -1,95 +1,60 @@
...
-            BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
-            // Get input steam
-            String inputLine;
-            StringBuilder response = new StringBuilder();
-            while ((inputLine = in.readLine()) != null) {
-                response.append(inputLine).append("\n");
-            }
-            // Close connection
-            in.close();
+            String url = "https://www.dhl.com/DatPublic/search.do?autoSearch=true&l=fi&directDownload=XML&statusHistory=true&search=consignmentId&a=" + parcelCode;
```
Comment (ristomatti) [inline app/src/main/java/com/nitramite/courier/DHLActiveTrackingStrategy.java:39]: Constant here also?

File: app/build.gradle [hunk 5]
```diff
@@ -57,6 +57,7 @@ dependencies {
     implementation 'com.google.android:flexbox:1.1.0'
     testImplementation 'junit:junit:4.12'
     implementation 'androidx.preference:preference:1.1.0'
+    implementation 'com.squareup.okhttp3:okhttp:4.8.0'
```
[the remark this answers is not in the document]
Comment (norkator) [reply app/build.gradle:60]: standard API

File: app/src/main/java/com/nitramite/courier/ArraPakettiStrategy.java (hunk 1 above)
Comment (norkator) [reply app/src/main/java/com/nitramite/courier/ArraPakettiStrategy.java]: Like this? https://github.com/norkator/Paketin-Seuranta/pull/18/commits/c2c3f429fc836d6a33ff68eddf661ee0a26ef0a0
I left those intact which parts are constructed like url + tracking number + some latter part

File: app/src/main/java/com/nitramite/courier/DHLActiveTrackingStrategy.java (hunk 4 above)
Comment (norkator) [reply app/src/main/java/com/nitramite/courier/DHLActiveTrackingStrategy.java:39]: Should I make part "https://www.dhl.com/DatPublic/search.do?autoSearch=true&l=fi&directDownload=XML&statusHistory=true&search=consignmentId&a=" private static final string? and then later append url + parcelCode? is that what you mean?

File: app/src/main/java/com/nitramite/courier/CainiaoStrategy.java (hunk 2 above)
Comment (norkator) [reply app/src/main/java/com/nitramite/courier/CainiaoStrategy.java]: https://github.com/norkator/Paketin-Seuranta/pull/18/commits/a42b595abf074ab349d304c8563c20f54c831ea7 how about now? same agent, tested to be working.

File: app/build.gradle (hunk 5 above)
[the remark this answers is not in the document]
Comment (ristomatti) [reply app/build.gradle:60]: Yes. Note that by convention Java constants should be all caps. :stuck_out_tongue:.

[the remark this answers is not in the document]
Comment (ristomatti) [reply app/build.gradle:60]: This comment should be for the next one. Mobile interface "somehow" put it here.

File: app/src/main/java/com/nitramite/courier/CainiaoStrategy.java (hunk 2 above)
Comment (ristomatti) [reply app/src/main/java/com/nitramite/courier/CainiaoStrategy.java]: Much cleaner!

File: app/src/main/java/com/nitramite/courier/DHLActiveTrackingStrategy.java (hunk 4 above)
Comment (ristomatti) [reply app/src/main/java/com/nitramite/courier/DHLActiveTrackingStrategy.java:39]: Didn't check so carefully but yep it'd more consistent.

**Comment by developerfromjokela** [comment]
At first glance looks good 😎.

**Pull request merged**